src/instrumentation/vertexai/patch.ts

```diff
@@ -0,0 +1,241 @@
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+/*
+ * Copyright (c) 2024 Scale3 Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { LANGTRACE_ADDITIONAL_SPAN_ATTRIBUTES_KEY } from '@langtrace-constants/common'
+import { calculatePromptTokens, estimateTokens } from '@langtrace-utils/llm'
+import { addSpanEvent, createStreamProxy } from '@langtrace-utils/misc'
+import {
+  Event,
+  LLMSpanAttributes,
+  Vendors
+} from '@langtrase/trace-attributes'
+import {
+  context,
+  diag,
+  Exception,
+  Span,
+  SpanKind,
+  SpanStatusCode,
+  trace,
+  Tracer
+} from '@opentelemetry/api'
+
+interface CandidateContentPart {
+  text: string
+}
+
+interface CandidateContent {
+  role: string
+  parts: CandidateContentPart[]
+}
+
+interface Candidate {
+  content: CandidateContent
+  finishReason: string
+  safetyRatings: any[] // You can replace 'any' with a more specific type if needed
+  avgLogprobs: number
+  index: number
+}
+
+interface UsageMetadata {
+  promptTokenCount: number
+  candidatesTokenCount: number
+  totalTokenCount: number
+}
+
+interface Response {
+  text?: any
+  candidates: Candidate[]
+  usageMetadata: UsageMetadata
+}
+
+interface Resp {
+  stream?: any
+  response: Response
+}
+
```

Comment:
Create a types.ts file in the same folder and move these there. You can look at `cohere` for an example

Reply:
moved